### PR TITLE
SC-17421 add kube-controller metrics and descriptions

### PR DESCRIPTION
**Pull request by yago-123**

### CODE REVIEW

File: docs/integration/kubernetes.md [hunk 1]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
 |----|-------|--------|-------------------------------------|
 |scheduler.attempts.total_num| gauge |  | number of attempts to schedule pods |
 
+### Kube-controller Metrics
+#### Workqueue Wait
+| Metric Name                              | Type  | Unit    | Description                                          |
+|------------------------------------------|-------|---------|------------------------------------------------------|
+| kubecontroller.workqueue_wait.p50latency | gauge | seconds | p50 latency of waiting time before item is processed |
```
Comment (otisg) [inline docs/integration/kubernetes.md:779]: latency of waiting time => wait time latency or simply wait latency

File: docs/integration/kubernetes.md [hunk 2]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+#### Workqueue Process
+|Metric Name| Type  | Unit    | Description                                            |
+|----|-------|---------|--------------------------------------------------------|
+|kubecontroller.workqueue_process.p50latency| gauge | seconds | p50 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p75latency| gauge | seconds | p75 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p90latency| gauge | seconds | p90 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p95latency| gauge | seconds | p95 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p99latency| gauge | seconds | p99 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.duration| gauge | seconds | duration of item being processed by Kube-controller    |
+|kubecontroller.workqueue_process.total_num| gauge |   | number items being processed by Kube-controller        |
```
Comment (otisg) [inline docs/integration/kubernetes.md:796]: number OF ...

File: docs/integration/kubernetes.md (hunk 1 above)
Comment (otisg) [inline docs/integration/kubernetes.md:779]: latency of waiting time => wait time latency or simply wait latency

File: docs/integration/kubernetes.md [hunk 3]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+
+#### Workqueue Depth
+|Metric Name| Type  | Unit    | Description                                 |
+|----|-------|---------|---------------------------------------------|
+|kubecontroller.workqueue_depth.total_num| gauge |   | number of items pending for being processed |
```
Comment (otisg) [inline docs/integration/kubernetes.md:801]: number of items queued for processing

File: docs/integration/kubernetes.md [hunk 4]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+
+#### Workqueue Unfinished Work
+|Metric Name| Type  | Unit    | Description                                         |
+|----|-------|---------|-----------------------------------------------------|
+|kubecontroller.workqueue_unfinished_work.total_num| gauge |   | number of items remaining unfinished in a workqueue |
```
Comment (otisg) [inline docs/integration/kubernetes.md:806]: number of items remaining in the work queue
or simply
number of items in the work queue

File: docs/integration/kubernetes.md [hunk 5]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+
+#### Workqueue Longest Running
+|Metric Name| Type  | Unit    | Description                                                 |
+|----|-------|---------|-------------------------------------------------------------|
+|kubecontroller.workqueue_longest_running.total_num| gauge |   | number of longest-running task processor within a workqueue |
```
Comment (otisg) [inline docs/integration/kubernetes.md:811]: number of longest-running task processors in work queue

File: docs/integration/kubernetes.md (hunk 4 above)
Comment (otisg) [reply docs/integration/kubernetes.md:806]: work queue or workqueue. I assume the former is better.  Just be consistent.

File: docs/integration/kubernetes.md [hunk 6]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+
+#### Unhealthy Nodes
+| Metric Name                               | Type  | Unit    | Description                                  |
+|-------------------------------------------|-------|---------|----------------------------------------------|
+| kubecontroller.unhealthy_nodes.percentage | gauge |   | number of unhealthy nodes in a specific zone |
+
+#### Workqueue Retries
+|Metric Name| Type  | Unit    | Description                                                   |
+|----|-------|---------|---------------------------------------------------------------|
+|kubecontroller.workqueue_retries.total_num| gauge |   | number of retries that have been attempted by workqueue items |
```
Comment (otisg) [inline docs/integration/kubernetes.md:831]: s/that have been//

**Review by otisg — CHANGES_REQUESTED** [review]

File: docs/integration/kubernetes.md [hunk 7]
```diff
@@ -772,6 +772,64 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+
+#### Workqueue Process
+|Metric Name| Type  | Unit    | Description                                            |
+|----|-------|---------|--------------------------------------------------------|
+|kubecontroller.workqueue_process.p50latency| gauge | seconds | p50 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p75latency| gauge | seconds | p75 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p90latency| gauge | seconds | p90 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p95latency| gauge | seconds | p95 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.p99latency| gauge | seconds | p99 latency of item being processed by Kube-controller |
+|kubecontroller.workqueue_process.duration| gauge | seconds | duration of item being processed by Kube-controller    |
```
Comment (cpipilas) [inline docs/integration/kubernetes.md:795]: >  duration of item being processed by Kube-controller

---> "Duration of processing an item from workqueue"

**Comment by yago-123** [comment]
Addressed the comments, GitHub still does not notify about this repo

**Review by cpipilas — APPROVED** [review]
LGTM

**Pull request merged**